### ISSUE TITLE
[java] ImmutableField: False positive on picocli annotated fields

### ISSUE THREAD

**Issue by zman0900** [pmd/pmd#6925]
**Affects PMD Version:** 7.26.0

**Rule:** ImmutableField

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/pmd/pmd_rules_java_design.html#immutablefield>

**Description:**
Non-final private field annotated by picocli's `@Option` annotation (and probably their `@Parameters` annotation too) violates the ImmutableField rule, but picocli will write such a field when the associated option is provided on the command line.  

See: <https://picocli.info/#_options_and_parameters>

**Code Sample demonstrating the issue:**

```java
public class Demo {
    @Option(
            names = "--ref-date",
            description = "Reference date (yyyy-MM-dd) for calculating"
    )
    private LocalDate refDate = LocalDate.now().withDayOfMonth(1);

    public void something() {...}
}
```

**Expected outcome:** PMD either ignores the ImmutableField rule when these specific picocli annotations are present, or provides some option to configure a custom set of annotations that should have the same effect.

PMD reports a violation at line 6, but that's wrong. That's a false positive.

**Running PMD through:** *Gradle*
